client/layout/index.js

```diff
@@ -15,8 +16,43 @@ import Header from 'layout/header';
 import { Controller, getPages } from './controller';
 import history from 'lib/history';
 import Notices from './notices';
+import { recordPageView } from 'lib/tracks';
 
 class Layout extends Component {
+	componentDidMount() {
+		this.recordPageViewTrack();
+	}
+
+	componentDidUpdate( prevProps ) {
+		const previousPath = get( prevProps, 'location.pathname' );
+		const currentPath = get( this.props, 'location.pathname' );
+
+		if ( ! previousPath || ! currentPath ) {
+			return;
+		}
+
+		if ( previousPath !== currentPath ) {
+			this.recordPageViewTrack();
+		}
+	}
+
+	recordPageViewTrack() {
+		const pathname = get( this.props, 'location.pathname' );
+		if ( ! pathname ) {
+			return;
+		}
+
+		// Remove leading slash, and camel case remaining pathname
+		let path = pathname.substring( 1 ).replace( /\//, '_' );
```

Comment:
We'll want the global flag, `/\//g`, for when we have `/analytics/products/<product_detail>`